src/epplibwrapper/errors.py

```diff
@@ -62,14 +62,17 @@ class RegistryError(Exception):
         - 2501 - 2502 Something malicious or abusive may have occurred
     """
 
-    def __init__(self, *args, code=None, note="", **kwargs):
+    def __init__(self, *args, code=None, note="", response=None, **kwargs):
         super().__init__(*args, **kwargs)
         self.code = code
+        self.response = response
         # note is a string that can be used to provide additional context
         self.note = note
 
     def should_retry(self):
-        return self.code == ErrorCode.COMMAND_FAILED
+        # COMMAND_USE_ERROR is returning with message, Registrar is not logged in,
+        # which can be recovered from with a retry
+        return self.code == ErrorCode.COMMAND_FAILED or self.code == ErrorCode.COMMAND_USE_ERROR
```

Comment:
@dave-kennedy-ecs  there are other times when this will throw a 2002, you need to check the .msg field as well not just the code. The main one we want to check for is "Registrar is not logged in." where as there is another .msg where it something like "command is not allowed"
```suggestion
        return self.code == ErrorCode.COMMAND_FAILED or (self.code == ErrorCode.COMMAND_USE_ERROR and self.msg=="Registrar is not logged in.")
```

obviously in the suggestion above you should use an enum

Reply:
done